src/chainstate/stacks/boot/contract_tests.rs

```diff
@@ -429,6 +432,908 @@ impl HeadersDB for TestSimHeadersDB {
+                ])
+            )
+            .unwrap()
+            .0
+            .to_string(),
+            "(err 2)".to_string()
+        );
+
+        assert_eq!(
+            env.execute_transaction(
+                (&USER_KEYS[1]).into(),
+                None,
+                POX_2_CONTRACT_TESTNET.clone(),
+                "stack-extend",
+                &symbols_from_values(vec![
+                    Value::UInt(9),
+                    POX_ADDRS[2].clone(),
+                ])
+            )
+            .unwrap()
+            .0,
+            execute(&format!(
+                "(ok {{ stacker: '{}, unlock-burn-height: {} }})",
+                Value::from(&USER_KEYS[1]),
+                Value::UInt(600 + 9 * 150),
+            ))
+        );
+
+        for cycle in 0..20 {
+            eprintln!("Cycle number = {}", cycle);
+            let empty_set = cycle < 1 || cycle >= 13;
+            let expected = if empty_set {
+                "(u0 u0)"
+            } else {
+                "(u1000000 u1)"
+            };
+            assert_eq!(
+                env.eval_read_only(
+                    &POX_2_CONTRACT_TESTNET,
+                    &format!("(list (default-to u0 (get total-ustx (map-get? reward-cycle-total-stacked {{ reward-cycle: u{} }})))
+                                    (default-to u0 (get len (map-get? reward-cycle-pox-address-list-len {{ reward-cycle: u{} }}))))",
+                             cycle, cycle))
+                    .unwrap()
+                    .0
+                    .to_string(),
+                expected
+            );
+            if !empty_set {
+                let expected_pox_addr = if cycle > 3 {
+                    &POX_ADDRS[2]
+                } else {
+                    &POX_ADDRS[1]
+                };
+                assert_eq!(
+                    env.eval_read_only(
+                        &POX_2_CONTRACT_TESTNET,
+                        &format!("(unwrap-panic (map-get? reward-cycle-pox-address-list {{ reward-cycle: u{}, index: u0 }}))",
+                                 cycle))
+                        .unwrap()
+                        .0,
+                    execute(&format!(
+                        "{{ pox-addr: {}, total-ustx: u{} }}",
+                        expected_pox_addr,
+                        1_000_000
+                    ))
+                );
+            }
+        }
+    });
+}
+
+#[test]
+fn pox_2_delegate_extend_units() {
+    let mut sim = ClarityTestSim::new();
+    sim.epoch_bounds = vec![0, 1];
+    let delegator = StacksPrivateKey::new();
+
+    // execute past 2.1 epoch initialization
+    sim.execute_next_block(|_env| {});
+    sim.execute_next_block(|_env| {});
+    sim.execute_next_block(|_env| {});
+
+    sim.execute_next_block_as_conn(|conn| {
+        test_deploy_smart_contract(conn, &POX_2_CONTRACT_TESTNET, &POX_2_TESTNET_CODE).unwrap()
+    });
+
+    sim.execute_next_block(|env| {
+        assert_eq!(
```

Comment:
are the different tests inside of this `execute_next_block` block meant to interact? if not, could they be separated to highlight that they are not interacting?